plugins/org.wso2.developerstudio.eclipse.esb.cloud/src/org/wso2/developerstudio/eclipse/esb/cloud/client/IntegrationCloudServiceClient.java

```diff
@@ -0,0 +1,288 @@
+import java.util.stream.Collectors;
+
+import org.apache.http.client.CookieStore;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.wso2.developerstudio.eclipse.esb.cloud.exceptions.CloudDeploymentException;
+import org.wso2.developerstudio.eclipse.esb.cloud.exceptions.HttpClientException;
+import org.wso2.developerstudio.eclipse.esb.cloud.exceptions.InvalidTokenException;
+import org.wso2.developerstudio.eclipse.esb.cloud.exceptions.NetworkUnavailableException;
+import org.wso2.developerstudio.eclipse.esb.cloud.model.Application;
+import org.wso2.developerstudio.eclipse.esb.cloud.resources.CloudServiceConstants;
+import org.wso2.developerstudio.eclipse.esb.cloud.resources.ResponseMessageConstants;
+import org.wso2.developerstudio.eclipse.esb.cloud.util.HTTPClientUtil;
+import org.wso2.developerstudio.eclipse.esb.cloud.util.JsonUtils;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Handles WSO2 integration cloud service actions
+ *
+ */
+public class IntegrationCloudServiceClient {
+
+    // maintains cookies that are required for authentication
+    private static CookieStore cookieStore;
+    private static IntegrationCloudServiceClient client;
+
+    private IntegrationCloudServiceClient() {
+
+    }
+
+    public static IntegrationCloudServiceClient getInstance() {
+        if (null == client) {
+            cookieStore = new BasicCookieStore();
+            client = new IntegrationCloudServiceClient();
+        }
+        return client;
+    }
+
+    /**
+     * Login to the Integration Cloud
+     * 
+     * @param username
+     * @param password
+     * @param tenant
+     * @return true if the user has successfully logged in, false if not
+     * @throws CloudDeploymentException
+     * @throws InvalidTokenException
+     * @throws HttpClientException
+     * @throws NetworkUnavailableException
+     */
+    public boolean login(String username, String password, String tenant)
+            throws CloudDeploymentException, InvalidTokenException, NetworkUnavailableException, HttpClientException {
+        String loginUrl = CloudServiceConstants.ServiceEndpoints.LOGIN_URL;
+
+        // Integration Cloud expects the username to be in <email>@<organization_key> format
+        String user = username + "@" + tenant;
+
+        Map<String, String> data = new HashMap<>();
+        data.put("action", CloudServiceConstants.Actions.LOGIN);
+        data.put("userName", user);
+        data.put("password", password);
+
+        String response = HTTPClientUtil.sendPostWithFormData(loginUrl, new HashMap<String, String>(), data,
+                cookieStore);
+
+        JsonParser parser = new JsonParser();
+        JsonElement jsonResponse = parser.parse(response);
+
+        String message = jsonResponse.getAsJsonObject().get("message").getAsString();
+        return message.equals(CloudServiceConstants.ResponseMessages.USER_SUCCESSFULLY_LOGGED_IN);
```

Comment:
Shall we switch the parameters for equals method to make sure it doesn't throw NPE?

Reply:
Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063